File: dc-02-19-19/consumer/src/main/java/com/example/demo/DemoClient.java
package com.example.demo;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;
import org.springframework.web.reactive.function.client.WebClient;
import reactor.core.publisher.Flux;

@Component
public class DemoClient {
  private final WebClient webClient;

  public DemoClient(WebClient client) {
    this.webClient = client;
  }

  @Value("${server.url:http://localhost:8090}")
  String baseUri;

  public Flux<Person> getAll() {
    return webClient.get().uri(baseUri + "/all").retrieve().bodyToFlux(Person.class);
  }
}

File: capetown-04-12-19/capetown-consumer/src/main/kotlin/com/demo/capetown/consumer/DemoClient.kt
package com.demo.capetown.consumer

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.MediaType
import org.springframework.stereotype.Component
import org.springframework.web.reactive.function.client.WebClient
import reactor.core.publisher.Flux
import java.time.Instant
import java.util.*

data class DemoUser(val id: UUID, val name: String, val modified: Instant)

@Component
class DemoClient(@Autowired
                 private val webClient: WebClient) {

    fun getAll(): Flux<DemoUser> {
        return webClient
                .get()
                .uri("http://localhost:8090/all")
                .accept(MediaType.APPLICATION_JSON_UTF8)
                .exchange()
                .flatMapMany {
                    it.bodyToFlux(DemoUser::class.java)
                }
    }
}